### PR TITLE
chore(deps): Update Go, cometbft, rocksdb dependencies

### PR DESCRIPTION
**Pull request by drklee3**
## Description
- [x] `cometbft` `v0.37.9-kava.1` -> [`v0.37.13-kava.1`](https://github.com/Kava-Labs/cometbft/commits/v0.37.13-kava.1/)
- [x] `cometbft-db` `v0.9.1-kava.2` -> [`v0.9.5`](https://github.com/cometbft/cometbft-db/commits/v0.9.5/) (upstream)
  - [Our modifications](https://github.com/Kava-Labs/cometbft-db/commits/kava-release-v0.9.1/) are no longer necessary:
    - Upstream updates to rocksdb `v9.2.1`, our RocksDB version modification `v1.8.6` -> `v1.8.13` no longer needed
    - Creation of RocksDB directory now handled via `opendb` module
      - `kava/opendb` module depends on compatible versions of `cometbft-db` and `grocksdb` so it doesn't need to be tagged with a new version
- [x] goleveldb replaced with [`informalsystems/goleveldb`](https://github.com/informalsystems/goleveldb/commits/v0.0.0/) fork to match cometbft-db
- [x] Go `1.21` -> `1.22`
  - CometBFT requires 1.22+
- [x] RocksDB `v8.10.0` -> `v9.3.1` via [`linxGnu/grocksdb@v1.9.3`](https://github.com/linxGnu/grocksdb/releases/tag/v1.9.3)
- [x] Update other `go.mod` files, e.g. e2e-ibc

---

- Pushed [`kava/rocksdb@v9.3.1-go1.22`](https://hub.docker.com/layers/kava/rocksdb/v9.3.1-go1.22/images/sha256-35dd66e6bc7d7753faeb79ad33882fca773d2d18be47891b3ddfd6026451abb1?context=explore) docker image

## Checklist
 - [x] Changelog has been updated as necessary.

### CODE REVIEW

**Comment by pirtleshell** [comment]
i believe a rebase of this branch onto `release/v0.27.x` will resolve all mint CI failures.

File: .github/workflows/ci-rocksdb-build.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: Continuous Integration (Rocksdb Build)
 
 env:
-  ROCKSDB_VERSION: v8.10.0
+  ROCKSDB_VERSION: v9.3.1
```
Comment (nddeluca) [inline .github/workflows/ci-rocksdb-build.yml:4]: Is 9.3.1 a typo or is 9.2.1 the typo?

Comment (drklee3) [reply .github/workflows/ci-rocksdb-build.yml:4]: ahh ty RocksDB `v9.3.1` should be the correct one that matches [`github.com/linxGnu/grocksdb@v1.9.3`](https://github.com/linxGnu/grocksdb/releases/tag/v1.9.3) which is the one used by kava as listed in `go.sum`

Got the rocksdb `v9.2.1` from  cometbft-db that depends on [`github.com/linxGnu/grocksdb@v1.9.2`](https://github.com/linxGnu/grocksdb/releases/tag/v1.9.2)

**Review by nddeluca — APPROVED** [review]
🔥 Tested locally against rocksdb 9.3.1, audited changes against v0.26.x, as well as audited changes in ethermint & cometbft

**Pull request merged**